agent/src/main/scala/io/scalac/mesmer/agent/akka/http/AkkaHttpAgent.scala

```diff
@@ -45,6 +45,15 @@ object AkkaHttpAgent
         .visit[HttpExtConnectionsAdvice]("bindAndHandle")
     )
 
+  def agent(config: com.typesafe.config.Config): Agent = {
+    def orEmpty(condition: Boolean, agent: Agent): Agent = if (condition) agent else Agent.empty
+    val configuration: AkkaHttpModule.Config             = module.enabled(config)
+
+    orEmpty(configuration.connections, connectionEvents) ++
+    orEmpty(configuration.requestTime, requestEvents) ++
+    orEmpty(configuration.requestCounter, requestEvents)
+  }
+
```

Comment:
Normally we use `LibraryInfo` to check if current setup allow for those libraries to instrumented - I wonder if we can guarantee that same level of safety with opentelemetry extension and config. If not this means that it would be possible to run transformations no supported classes and possibly breaking application.

Reply:
I might need more data on this one: 

 - why do we require (for example for http) akka-actor && akka-actor-typed being present on the classpath when we instrument akka http? Why can't we just require akka-http only?
 -  why is this mechanism "opt-in" rather than "opt-out"? The guts of the libs rarely change so we might be supporting only a subset of what we could support right now. In case we're opt out we could have a list of known problematic versions and skip them/instrument them differently.
 
 
When it comes to how open telemetry agent handles that, based on otel's `TypeInstrumentation.java` interface: 
 - it searches the classpath for the desired class (the `typeMatcher()` method). If it's able to find it, it attaches the code to it based on the instructions passed in `transform()` method
 - in case we're worried about the performance of the above, we can short-circuit the process early using `classLoaderOptimization()`
 - in case this is not enough and users will face some problems with a particular instrumentations, we can: turn off a lib in config/turn it off by default (explaining in the docs why)